cwl/src/main/scala/cwl/Workflow.scala

```diff
@@ -87,29 +86,33 @@ case class Workflow private(
           womTypeForInputParameter(i).map(i.id -> _).toList
         }.toMap
 
-    val graphFromInputs: Set[ExternalGraphInputNode] = inputs.map {
-      case inputParameter if inputParameter.default.isDefined =>
-        val parsedInputId = FileAndId(inputParameter.id).id
-        val womType = womTypeForInputParameter(inputParameter).get
-
-        val womValue = inputParameter.default.get.fold(InputParameter.DefaultToWomValuePoly).apply(womType).toTry.get
-        OptionalGraphInputNodeWithDefault(WomIdentifier(parsedInputId, inputParameter.id), womType, ValueAsAnExpression(womValue), parsedInputId)
-      case input =>
-        val parsedInputId = FileAndId(input.id).id
-
-        RequiredGraphInputNode(WomIdentifier(parsedInputId, input.id), womTypeForInputParameter(input).get, parsedInputId)
+    val externalGraphInputNodes: Set[ExternalGraphInputNode] = inputs.map { wip =>
+      val womType = wip.`type`.head.fold(MyriadInputTypeToWomType)
+      val parsedInputId = FileAndId(wip.id).id
+      val womId = WomIdentifier(parsedInputId, wip.id)
+
+      def optionalWithDefault(memberType: WomType): OptionalGraphInputNodeWithDefault = {
+        val defaultValue = wip.default.get.fold(InputParameter.DefaultToWomValuePoly).apply(womType).toTry.get
```

Comment:
I see that default is always defined but is `DefaultToWomValuePoly` result always going to be there?  If not can this whole thing become a `Checked`?